### ISSUE TITLE
Not working with Kerio Connect on macOS

### ISSUE THREAD

**Issue by mjungowski** [jobisoft/TbSync#204]
tbsync works great with our Kerio Connect server on Windows and Linux but reproducibly fails to connect on macOS, both 10.14 and 10.13. I get the dreaded "Communication error (HTTP status 400)" even though I have "Enforce provisioning (required by Kerio)" enabled.

However, I've noticed that as soon as I close the window the checkbox next to "Enforce provisioning" is unchecked again. Is it possible that for some obscure reason this checkbox isn't working on macOS?

**Comment by jobisoft** [comment]
Can you retry with latest beta?
https://tbsync.jobisoft.de/TbSyncBetaInstaller.xpi

**Comment by marjungo** [comment]
Thanks. I've had to manually reenable the EAS provider (the beta only came with a CalDAV/CardDAV beta provider) but now I'm getting "JavaScript error! Check debug log for details" as soon as I check the "Enforce provisioning" setting under Sychronisation settings.

Is it possible that I need a beta of the EAS provider as well?

**Comment by jobisoft** [comment]
Yes! The above installer should give you the beta version of all three modules, marked with [beta release channel] in the AddOn list.

![image](https://user-images.githubusercontent.com/5830621/50351938-f998d000-0543-11e9-896a-563fa0b82db3.png)

**Comment by marjungo** [comment]
Well, it didn't ;-)

No worries though, I found it on your website, removed the previously installed one, installed the beta provider and am now getting the same Communication error (HTTP status 400) again regardless of whether the "Enforce provisioning" box is checked or not.

Any logs I could provide you with?

**Comment by jobisoft** [comment]
You can prepare a bug report at "Help" in the Manager. You first need to enable debug logging there, restart TB (!), than toggle that Kerio option and create a bug report.

Does any other option show the same "i do not like to be changed" behaviour?

**Comment by mjungowski** [comment]
I've just tried the new versions and yes, it's still the same. I still keep getting the HTTP status 400 error message and the "Enforce provisioning" checkbox does not want to remain checked and is disabled again as soon as I click away from that pane.

**Comment by jobisoft** [comment]
This will take longer, I have no idea what macOS is doing there.

Is it only the checkboxes? Does the dropdown keep its new value?
Do you know how to open the Thunderbird Error Console? Tools -> Developer Tools -> Error Console.

Does anything show up there, when you try to toggle the setting?

**Comment by mjungowski** [comment]
Both dropdown menus seem to exhibit the same behavior, they also revert back to their default setting.

ActiveSync version reverts back to "Best available"
Separator for multiline address field reverts back to "Line break"

I've checked the Error Console but other than a warning, that my server still uses an SHA-1 certificate (which I am aware of, didn't have the time to replace it just yet) there are no log entries in there at all when I check/uncheck these boxes.

**Comment by jobisoft** [comment]
i have feedback from other mac users, that this is not a general error. Can you create a fresh profile (thunderbird -p) and setup tbsync there again? The Beta? Same behaviour?

**Comment by mjungowski** [comment]
Interesting... I have just created this profile from scratch, first time I ever launched Thunderbird on this Mac. Me and my colleague were also able to reproduce this every time on four different Macs: a MacBook Pro running 10.14, a MacBook running 10.14, an iMac running 10.14 and another one running 10.13.

Just to make sure I did it again, deleted my profile, created a new one, and installed the Beta extension but the behavior is absolutely identical. No difference at all.

**Comment by jobisoft** [comment]
Hm, you had this behavior with the stable releases from ATN as well, right? So it is not something I have introduced in the beta.

The only option we have is, that I build special version for you to try out, to pinpoint the error. Are you avail the next couple of days?

**Comment by mjungowski** [comment]
Correct, I've noticed this behavior with the stable releases as well and it's definitely not unique to the beta.

I should be available on and off, not 24/7 but most of the time for sure. Let me know if you need me to do or test anything.

**Comment by jobisoft** [comment]
The latest beta adds some debug messages to the new error log:

![image](https://user-images.githubusercontent.com/5830621/50382845-8ceb1600-06a8-11e9-9fae-d07870fd235e.png)

Do you see either of the following Messages, when you toggle the kerio Setting?
 - onblur method for instant save of <tbsync.accountsettings.pref.provision> called
 - onblur eventlistener for instant save of <tbsync.accountsettings.pref.provision> called

After you toggled the kerio option, open the error log here:

![image](https://user-images.githubusercontent.com/5830621/50382861-d50a3880-06a8-11e9-8259-743bbb16b910.png)

To get the latest beta, just update your add ons in the AddOn Manager:

![image](https://user-images.githubusercontent.com/5830621/50382886-69749b00-06a9-11e9-8b7e-baab160cc65b.png)

**Comment by mjungowski** [comment]
I do see those errors, yes.

<img width="712" alt="screen shot 2018-12-26 at 2 21 08 pm" src="https://user-images.githubusercontent.com/46053063/50456373-8c0ceb00-0919-11e9-937c-f71196111cb8.png">

**Comment by jobisoft** [comment]
This is very strange. There is no difference between pref.downloadonly and pref.provision. They are both checkboxes. Can you please update your beta (TbSync should update to 1.0.14) and play again with the kerio option and check the log?

I also added a click event:

![image](https://user-images.githubusercontent.com/5830621/50496091-ca93ca00-0a2d-11e9-8e04-18852cc9cc38.png)

**Comment by mjungowski** [comment]
Yes, I can see the onclick eventlistener in the log every time I click it. Here are the steps I took in order to create the event log attached at the bottom:

- Opened TbSync account manager by clicking the TbSync text at the bottom right in Thunderbird's status bar
- 11:29:05 AM: Clicked on synchronization settings
- 11:29:20 AM: Clicked onto another window, thereby taking away focus from the TbSync account manager window which produced the first two debug log entries
- 11:29:38 AM: Clicked on "Revert local changes" to activate it
- 11:29:42 AM: Clicked on "Enforce provisioning" to activate it
- 11:29:46 AM: Clicked onto another window again

The thing I've noticed is that the event seems to register, however you seem to trigger another event whenever the TbSync account manager window loses focus. I can see an eventlistener and a method for <tbsync.accountsettings.pref.downloadonly> but not for <tbsync.accountsettings.pref.provision>. It either doesn't register that the provision checkbox was changed, or it can't read its value, or it simply doesn't trigger the event for that particular checkbox for some odd reason.

<img width="712" alt="screen shot 2018-12-28 at 11 30 38 am" src="https://user-images.githubusercontent.com/46053063/50523175-3c066380-0a94-11e9-8196-932ba6ef20b7.png">
